File: leet_DP_0001_two_sum.js
const twoSum = function(nums, target)
{
    const mp = {};
    
    for (let i = 0; i < nums.length; i++)
    {
        if (mp[ nums[i] ] >= 0)
            return [ mp[nums[i] ] , i] 
        mp[ target - nums[i] ] = i
    }
    return {}
};

/**
 * @param {number[]} nums
 * @param {number} target
 * @return {number[]}
 */

File: leet_DP_0001_two_sum.py
#* ******************************************** */
#*                                              */
#*              ~  ~  ~  ~  ~    ~              */
#*               ~    _ ~ _   o>                */
#*              ~  \ / \ / \ /  ~               */
#*               ~  ~      ~    ~               */
#*                                              */
#* ******************************************** */

class Solution:
    def twoSum(self, nums: List[int], target: int) -> List[int]:
        M = {}
        for i in range(len(nums)):
            n = nums[i]
            if target - n in M:
                break
            M[n] = i
        return [M[target - n], i]
